crates/ty_python_semantic/src/types/builder.rs

```diff
@@ -780,6 +780,37 @@ impl<'db> IntersectionBuilder<'db> {
                     seen_aliases,
                 )
             }
+            Type::EnumLiteral(enum_literal) => {
+                let enum_class = enum_literal.enum_class(self.db);
+                let metadata =
+                    enum_metadata(self.db, enum_class).expect("Class of enum literal is an enum");
+
+                let enum_members_in_negative_part = self
+                    .intersections
+                    .iter()
+                    .flat_map(|intersection| &intersection.negative)
+                    .filter_map(|ty| ty.as_enum_literal())
+                    .filter(|lit| lit.enum_class(self.db) == enum_class)
+                    .map(|lit| lit.name(self.db).clone())
+                    .chain(std::iter::once(enum_literal.name(self.db).clone()))
+                    .collect::<FxOrderSet<_>>();
```

Comment:
I think this can be an `FxHashSet`, which is a slighlty simpler datastructure and possibly slightly faster? I don't think deterministic ordering is necessary here

```suggestion
                    .collect::<FxHashSet<_>>();
```